tests/integration-tests/tests/storage/snapshots_factory.py

```diff
@@ -0,0 +1,255 @@
+        :param request: The current request
+        :param subnet_id: The subnet id where to get the snapshot
+        :param region: The region where to get the snapshot
+        """
+        # Only one snapshot creation per factory allowed
+        if self.snapshot is not None:
+            raise Exception("Snapshot already created")
+
+        self.ec2 = boto3.resource("ec2", region_name=region)
+        self.boto_client = boto3.client("ec2", region_name=region)
+
+        cluster_config = SnapshotConfig(
+            request.config.getoption("key_path"),
+            request.config.getoption("key_name"),
+            self.ec2.Subnet(subnet_id).vpc_id,
+            subnet_id,
+        )
+        self.snapshot = self._create_snapshot(region, cluster_config)
+        return self.snapshot.id
+
+    def _create_snapshot(self, region, cluster_config):
+        self.config = cluster_config
+        ami_id = self._get_amazonlinux_ami()
+
+        self.security_group_id = self._get_security_group_id()
+
+        subnet = self.ec2.Subnet(self.config.master_subnet_id)
+
+        # Create a new volume and attach to the instance
+        self.volume = self._create_volume(subnet)
+        self.instance = self._launch_instance(ami_id, subnet)
+        self._attach_volume()
+        # Open ssh connection
+        self.ssh_conn = self._open_ssh_connection()
+
+        # Partitions the disk with a gpt table and 1 single partition inside
+        self._format_volume(self.ssh_conn)
+
+        # Stops the instance before taking the snapshot
+        self._release_instance()
+
+        self.snapshot = self._create_volume_snapshot()
+        return self.snapshot
+
+    def _create_volume_snapshot(self):
+        logging.info("creating snapshot...")
+        snapshot = self.ec2.create_snapshot(Description="parallelcluster-test-snapshot", VolumeId=self.volume.id)
+        while snapshot.state == u"pending":
+            time.sleep(10)
+            snapshot = self.ec2.Snapshot(snapshot.id)
+        logging.info("Snapshot ready: %s" % snapshot.id)
+        return snapshot
+
+    def _format_volume(self, ssh_conn):
+        logging.info("Partitioning device...")
+        ssh_conn.run("sudo sh -c 'echo -e \"g\nn\np\n1\n\n\nw\" | fdisk /dev/sdf'", warn=True, pty=False, hide=False)
+        # Finds out the device name of the volume
+        logging.info("Finding device name...")
+        device_name = ssh_conn.run("readlink -f /dev/sdf").stdout.strip()
+        # formats the 1st partition of disk
+        logging.info("Formatting 1st partition...")
+        ssh_conn.run("sudo sh -c 'mkfs.ext4 {}1'".format(device_name))
+        logging.info("Mounting partition...")
+        ssh_conn.run("sudo mkdir /mnt/tmp")
+        ssh_conn.run("sudo mount {}1 /mnt/tmp".format(device_name))
+        logging.info("Writing test data...")
+        ssh_conn.run("echo 'hello world' | sudo tee -a /mnt/tmp/test.txt")
+        logging.info("Device redy")
+
+    def _open_ssh_connection(self):
```

Comment:
Could we postpone this change to a future commit? I see the clusters_factory is now sleeping 60 seconds before returning the cluster and I think this is due to the connection errors that I am catching here while the instance is starting. So I would like to export this behaviour to the remote_command_executor and remove the 60 seconds sleep from clusters_factory but of course this would be better addressed as a separate fix. What do you think?

Reply:
the sleep of 60 seconds was not related to ssh failing to connect if I recall correctly. But yes we can postpone this